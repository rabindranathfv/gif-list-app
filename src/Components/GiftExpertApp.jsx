import React, {useState} from 'react';
import { AddCategory } from './AddCategory';
import { GiftGrid } from './GiftGrid';

const GiftExpertApp = () => {

    const [category, setCategory] = useState('');

    console.log('CATEGORY IN MAIN COMPONENT', category);

    return (
        <div>
            <h2> GifExpertAPP </h2>
            <AddCategory setCategory={setCategory} />
            <hr />
            <GiftGrid category={category}  />
        </div>
    )
}

GiftExpertApp.defaultProps = {
    category: 'dragon ball',
}

export default GiftExpertApp
