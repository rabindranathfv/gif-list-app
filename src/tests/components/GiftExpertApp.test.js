import { shallow } from 'enzyme';
import React from 'react';
import GiftExpertApp from '../../Components/GiftExpertApp';

describe('GiftExpertApp', () => {
    let wrapper = shallow( <GiftExpertApp />);

    beforeEach( () => {
        jest.clearAllMocks();
        wrapper = shallow( <GiftExpertApp />)
    })

    test('should render GiftExpertApp', () => {
        expect( wrapper).toMatchSnapshot();
    })
    

})
