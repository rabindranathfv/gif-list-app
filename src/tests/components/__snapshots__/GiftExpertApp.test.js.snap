// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GiftExpertApp should render GiftExpertApp 1`] = `
<div>
  <h2>
     GifExpertAPP 
  </h2>
  <AddCategory
    setCategory={[Function]}
  />
  <hr />
  <GiftGrid
    category={
      Array [
        "dragon ball",
      ]
    }
    key="dragon ball"
  />
</div>
`;
